### PR TITLE
[client] Print summary at cmd diff

### PR DESCRIPTION
**Pull request by csordasmarton**
Print a similar summary at the end of `CodeChecker cmd diff` command as at the `CodeChecker parse` command.

### CODE REVIEW

File: web/tests/functional/diff_local_remote/test_diff_local_remote.py [hunk 1]
```diff
@@ -207,14 +207,14 @@ def test_local_cmp_filter_unres_severity(self):
     def test_local_cmp_filter_unres_filepath(self):
         """Filter unresolved results by file path."""
         res = self.get_local_remote_diff(['--file', '*divide_zero.cpp'])
-        self.assertEqual(len(re.findall(r'divide_zero.cpp', res)), 4)
-        self.assertEqual(len(re.findall(r'new_delete.cpp', res)), 0)
+        self.assertEqual(len(re.findall(r'/divide_zero.cpp', res)), 4)
```
Comment (gyorb) [inline web/tests/functional/diff_local_remote/test_diff_local_remote.py]: why do we need these changes here, because of the summary printing the results twice?

File: web/client/codechecker_client/cmd_line_client.py [hunk 2]
```diff
@@ -873,6 +904,7 @@ def report_to_html(client, reports, output_dir):
                 checked_file, output_path))
 
         html_builder.create_index_html(output_dir)
+        print_stats(reports, file_stats, severity_stats)
```
Comment (gyorb) [inline web/client/codechecker_client/cmd_line_client.py]: Wouldn't be better to generate summary.html for the html output?

File: web/client/codechecker_client/cmd_line_client.py [hunk 3]
```diff
@@ -954,13 +991,19 @@ def print_reports(client, reports, output_format):
...
+            file_stats[file_name] += 1
+
         if output_format == 'plaintext':
             for row in rows:
                 print("[{0}] {1}: {2} [{3}]\n{4}\n".format(
                     row[0], row[1], row[2], row[3], row[4]))
         else:
             print(twodim_to_str(output_format, header, rows))
 
+        print_stats(reports, file_stats, severity_stats)
```
Comment (gyorb) [inline web/client/codechecker_client/cmd_line_client.py]: Do we print these stats after the json output too?

File: web/client/codechecker_client/cmd_line_client.py (hunk 2 above)
Comment (csordasmarton) [reply web/client/codechecker_client/cmd_line_client.py]: We already implemented this feature. See #2036.

File: web/tests/functional/diff_local_remote/test_diff_local_remote.py (hunk 1 above)
Comment (csordasmarton) [reply web/tests/functional/diff_local_remote/test_diff_local_remote.py]: Yes, you are right, because of the summary.

```sh
[INFO 2019-06-06 09:10] - Matching runs: simple
[MEDIUM] /home/username/projects/simple/main.cpp:3:3: expression result unused [clang-diagnostic-unused-value]
  sizeof(42);


[HIGH] /home/username/projects/simple/main.cpp:3:3: suspicious usage of 'sizeof(K)'; did you mean 'K'? [bugprone-sizeof-expression]
  sizeof(42);



----==== Summary ====----
-----------------------
Filename | Report count
-----------------------
main.cpp |            2
-----------------------
-----------------------
Severity | Report count
-----------------------
HIGH     |            1
MEDIUM   |            1
-----------------------
----=================----
Total number of reports: 2
----=================----

[INFO 2019-06-06 09:10] - Compared local report directory /home/username/projects/simple/reports and remote run(s) simple (matching: simple).
```

Instead of this I change the expected result count. This way we test the summary too.

File: web/client/codechecker_client/cmd_line_client.py (hunk 3 above)
Comment (csordasmarton) [reply web/client/codechecker_client/cmd_line_client.py]: No, we do not print the statistics after the json output. The json output handled in a different branch:
https://github.com/Ericsson/codechecker/blob/427c1ff3aa12c01c1e9ac4115e1cc91ae37d344f/web/client/codechecker_client/cmd_line_client.py#L884-L893

File: web/tests/functional/diff_local_remote/test_diff_local_remote.py [hunk 4]
```diff
@@ -207,14 +207,14 @@ def test_local_cmp_filter_unres_severity(self):
     def test_local_cmp_filter_unres_filepath(self):
         """Filter unresolved results by file path."""
         res = self.get_local_remote_diff(['--file', '*divide_zero.cpp'])
-        self.assertEqual(len(re.findall(r'divide_zero.cpp', res)), 4)
+        self.assertEqual(len(re.findall(r'divide_zero.cpp', res)), 5)
```
Comment (gyorb) [inline web/tests/functional/diff_local_remote/test_diff_local_remote.py]: Please add a comment here that in the output the file names are printed twice because of the summary.
It will be really hard to follow, how many reports are actually found and printed.

File: web/client/codechecker_client/cmd_line_client.py (hunk 2 above)
Comment (gyorb) [reply web/client/codechecker_client/cmd_line_client.py]: If we generate a html summary too, I do not see the extra benefit of printing it here too, but ok.
